src/lib/ProjectM36/Client/Simple.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module ProjectM36.Client.Simple (
+  simpleConnectProjectM36,
+  withTransaction,
+  execute,
+  query,
+  rollback,
+  close,
+  Atom(..),
+  AtomType(..),
+  DBError(..),
+  Attribute(..),
+  C.ConnectionInfo(..),
+  C.PersistenceStrategy(..),
+  C.NotificationCallback,
+  C.emptyNotificationCallback,
+  C.DatabaseContextExpr(..),
+  C.RelationalExprBase(..)  
+  ) where
+-- | A simplified client interface for Project:M36 database access.
+import ProjectM36.Error
+import ProjectM36.Base
+import qualified ProjectM36.Client as C
+import Control.Monad.Reader
+import Control.Exception.Base
+
+type DBConn = (C.SessionId, C.Connection)
+
+newtype DB a = DB {runDB :: ReaderT (C.SessionId, C.Connection) IO a}
+  deriving (Functor, Applicative, Monad)
+           
+newtype TransactionCancelled = TransactionCancelled DBError deriving Show
+instance Exception TransactionCancelled
+
+-- | A simple alternative to 'connectProjectM36' which includes simple session management
+simpleConnectProjectM36 :: C.ConnectionInfo -> IO (Either DBError DBConn)
+simpleConnectProjectM36 connInfo = do
+  eConn <- C.connectProjectM36 connInfo
+  case eConn of
+    Left err -> pure (Left (ConnError err))
+    Right conn -> do
+      eSess <- C.createSessionAtHead conn "master"
+      case eSess of
+        Left err -> do
+          C.close conn
+          pure (Left (RelError err))
+        Right sess -> pure (Right (sess, conn))
+        
+close :: DBConn -> IO ()        
+close (_ , conn) = C.close conn
+
+-- | Runs an IO monad which may include some database updates. If an exception or error occurs, the transaction is rolled back. Otherwise, the transaction is committed.
+withTransaction :: DBConn -> DB a -> IO (Either DBError a)
+withTransaction (sess, conn) dbm = do
+  eHeadName <- C.headName sess conn
+  case eHeadName of 
+    Left err -> pure (Left (RelError err))
+    Right headName -> do
+      let successFunc = C.autoMergeToHead sess conn UnionMergeStrategy headName
```

Comment:
Would it ever make sense to use a different strategy? If so, you could have `withTransactionUsing :: MergeStrategy -> ...` and then `withTransation = withTransactionUsing UnionMergeStrategy` or something. Maybe it doesn't make sense.